app/Actions/Album/SetProtectionPolicies.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\DTO\AlbumProtectionPolicies;
+use App\Exceptions\Internal\FrameworkException;
+use App\Exceptions\InvalidPropertyException;
+use App\Exceptions\ModelDBException;
+use App\Models\Extensions\BaseAlbum;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Support\Facades\Hash;
+
+/**
+ * Class SetProtectionPolicies.
+ */
+class SetProtectionPolicies extends Action
```

Comment:
Why do we suddenly have a `SetProtectionPolicies` what happened to `SetProtectionPolicy`?

Reply:
Just renamed it. Literally same file.

Reply:
Yes, but why?! In particular, GIT does not recognize this as a renaming which means that there has been too many changes and this makes it really hard to compare the file. (The same holds for some other files which you renamed.)

I would be in favor of the renaming if the previous name had been a misnomer, but in this case I do not even understand the reason. Why did you convert the name to its plural form? (Again this file and some other files as well.) An object of this class represents a _single_ access policy the same way as on object of `User` represents a single user or an object of `image` represents a single image. IMHO, the plural form is worse than before and additionally inconsistent with the naming scheme of the remaining classes.

Moreover, I hoped that I was nearly finished with the review. Now, I see 86 changed files since my last review again. I don't feel like re-re-re-re-re-reviewing everything again. I thought you said you were finished with this PR and it was ready for review. IMHO, this implies that there won't be any fundamental changes anymore, except for bug fixes and change requests of course. I am the last person who has a right to complain about huge PRs. Most of my PRs have been huge, too. @kamil4 can tell you a thing or two about it. But I always tried to not change a PR fundamentally after a review had started. I cannot even easily check whether my change requests have been addressed or how the have been addressed as those changes disappear in the additional changes.

Reply:
> > Yes, but why?!
> 
> Because for me it groups multiple policy:
> 
>     * Policy for download.
>     * Policy for visibility (hidden/visible).
>     * Policy for access (locked/unlocked).

IMHO, that is debatable. Surely, one could say these are different policies and the class should have been named in the plural form. But one could also say that these are only individual attributes and like a law or regulation which consists of different paragraphs a policy consists of a set of rules (determined by those attributes above) which work together and jointly constitute the policy. Calling each trivial boolean flag a policy on its own seems questionable. That's why I came up with the name "protection" policy in the first place, because I was looking for a term that covers all aspects.

Anyway, if you don't mind I would rather prefer to undo that change. It really makes reviewing this PR much more complicated than it already is.

Reply:
Undone: [3d5bb33](https://github.com/LycheeOrg/Lychee/pull/1539/commits/3d5bb335547e3be785a45f2cf14e7a313f5cb76f)